packages/pwa-kit-extension-sdk/src/express/middleware/apply-application-extensions.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Third-Party Imports
+import {Request, Response, NextFunction, Application} from 'express'
+
+// Local Imports
+import {getApplicationExtensions} from '../placeholders/application-extensions'
+
+// Define the middleware function that modifies the app
+const applyApplicationExtensions = (app: Application) => {
```

Comment:
An important thing to document for the server extension is that it must be idempotent. The server side hot reloading feature we have simply re-require the module, if the `extendApp` have side effects, that will break hot reloading.